Userland/Libraries/LibGfx/QOIWriter.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2022, Olivier De Cannière <olivier.decanniere96@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "QOIWriter.h"
+#include <AK/String.h>
+
+namespace Gfx {
+
+ByteBuffer QOIWriter::encode(Bitmap const& bitmap)
+{
+    QOIWriter writer;
+    writer.add_header(bitmap.width(), bitmap.height(), Channels::RGBA, Colorspace::sRGB);
+
+    Color previous_pixel = { 0, 0, 0, 255 };
+
+    bool creating_run = false;
+    int run_length = 0;
+
+    for (int y = 0; y < bitmap.height(); y++) {
+        for (int x = 0; x < bitmap.width(); x++) {
+            Color pixel = bitmap.get_pixel(x, y);
+
+            // Check for at most 62 consecutive identical pixels
+            if (pixel.red() == previous_pixel.red() && pixel.green() == previous_pixel.green() && pixel.blue() == previous_pixel.blue() && pixel.alpha() == previous_pixel.alpha()) {
```

Comment:
`Color` has `operator==` :)

```suggestion
            if (pixel == previous_pixel) {
```